R/load-code.r

```diff
@@ -3,51 +3,45 @@
 #' Load all R code in the `R` directory. The first time the code is
 #' loaded, `.onLoad` will be run if it exists.
 #'
-#' @param pkg package description, can be path or package name.  See
-#'   [as.package()] for more information
+#' @inheritParams load_all
 #' @keywords programming
 #' @export
-load_code <- function(pkg = ".") {
-  pkg <- as.package(pkg)
-  env <- ns_env(pkg)
+load_code <- function(path = ".") {
+  path <- pkg_path(path)
+  package <- pkg_name(path)
+  encoding <- pkg_desc(path)$get("Encoding")
 
-  r_files <- find_code(pkg)
+  env <- ns_env(package)
+
+  r_files <- find_code(path)
   paths <- changed_files(r_files)
   if (length(paths) == 0L) return()
 
   success <- FALSE
   cleanup <- function() {
     if (success) return()
     clear_cache()
-    unload(pkg)
+    unload(package)
```

Comment:
I changed unload to take a package name rather than a path. So it is correct as written, but we can discuss whether that API is the right one.